validate/json_diff_test.py

```diff
@@ -167,16 +252,43 @@ def test_ignore(self):
                 }))
         )
 
+    def test_duplicates(self):
+        self.assertEqual(
+            {
+                'foo': {
+                    'b': {'key': 'b', 'value': 'y'},
+                    'a': {'key': 'a', 'value': 'q'},
+                    'c': {'key': 'c', 'value': 'n'}
+                },
+                'bar': 'baz'
+            },
+            prepare_diff_input(
+                {
+                    'foo': [{'key': 'b', 'value': 'y'},
+                            {'key': 'c', 'value': 'n'},
+                            {'key': 'c', 'value': 'm'},
+                            {'key': 'a', 'value': 'q'}],
+                    'bar': 'baz'
+                },
+                DiffConfig(children={
+                    'foo':
+                        DiffConfig(
+                            primary_key='key'
+                        )
+                })))
+
 
```

Comment:
Thanks for adding the test cases. Can you also add test cases where there is counter difference and for comparing 2 directories

Reply:
The counter logic is tested in JsonDiffTest. I added hard-coded counters value in the tests. (I'm not sure what do you mean by counter difference because there is only one counters for each diff).

Comparing 2 dirs is part of tasks.py which I think it's OK not to test because it's basically thin wrapper of json_diff functions (under the standard of current WM codebase :))

Reply:
ok nevermind. The counters are for each type, so its checking the test case I was talking about.